widget/view.controller.js

```diff
@@ -366,49 +409,128 @@
+                                                            if (data.status === "Available") {
+                                                                $scope.connectorHealthStatus[index] = true;
+                                                            }
+                                                        });
+                                                } else {
+                                                    $scope.isConnectorConfigured = false;
+                                                }
+                                            });
+                                    });
+                            } else {
+                                console.error('No data found in response[\'hydra:member\']');
+                            }
+                        })
+                        .catch(function (error) {
+                            console.error('An error occurred:', error);
+                        });
+                    // Add the promise to the array
+                    promises.push(promise);
+                }
+                // Use Promise.all to wait for all promises to complete
+                Promise.all(promises)
+                    .then(() => {
+                        // After all promises are resolved, evaluate the condition
+                        $scope.isConnectorsHealthy = false;
+                        let indices = _.map(_.filter($scope.connectorHealthStatus, value => value === false), (value, index) => $scope.connectorHealthStatus.indexOf(value, index));
+                        const notConfigConnectors = _.uniq(indices).map(index => $scope.selectedEnv.huntTools[index]);
+                        const toasterMessage = 'Connector ' + notConfigConnectors.join(', ') + ' is not configured';
+                        if (notConfigConnectors.length === 0) {
+                            $scope.selectedEnv.installOutbreakType = $scope.outbreakAlertSeverityList.slice();
+                            WizardHandler.wizard('OutbreaksolutionpackWizard').next();
+                        } else {
+                            var huntToolIndex = $scope.selectedEnv.huntTools.indexOf(notConfigConnectors[0]);
+                            $scope.params.activeTab = huntToolIndex;
+                            loadActiveTab(huntToolIndex, notConfigConnectors[0]);
+                            var connectorConfig = document.getElementById('accordion-connector-config-' + huntToolIndex);
+                            connectorConfig.childNodes[2].classList.add('in');
+                            toggleConnectorConfigSettings(huntToolIndex);
+                            toaster.error({
+                                body: toasterMessage
+                            });
+                        }
+                    })
+                    .catch(error => {
+                        console.error('An error occurred in Promise.all:', error);
```

Comment:
An error occurred in Promise.all:' --> what does this error means to the user?

Reply:
runtime error will catch by try and saved in error var, so depends on error it will populate the error